alert_system/tasks.py

```diff
@@ -0,0 +1,50 @@
+import logging
+
+from celery import shared_task
+from celery.exceptions import MaxRetriesExceededError
+from django.db import transaction
+from django.utils import timezone
+
+from alert_system.mappings import CONNECTOR_REGISTRY
+
+from .models import Connector
+
+logger = logging.getLogger(__name__)
+
+
+@shared_task(bind=True, max_retries=3, retry_backoff=True)
+def process_connector_task(self, connector_id):
+    """
+    Celery task to run ETL for one connector.
+    """
+    connector = Connector.objects.get(id=connector_id)
+    connector.status = Connector.Status.RUNNING
+    connector.save(update_fields=["status"])
+    connector_class = CONNECTOR_REGISTRY[connector.type]
+    extraction_class = connector_class.extractor
```

Comment:
We can do smth like this???

```suggestion
    extraction_class = connector.get_extractor_class()
    if not connector_class:
        logger.error(f"[ETL] No connector class found for connector type: {connector.type}")
        return

    ----###In connector model###----
    
        def get_extractor_Class(self):
        from alert_system.etl.Gdacs_flood.extraction import GdacsFloodExtraction

        match self.ConnectorType(self.type):
            case self.ConnectorType.GDACS_FLOOD:
                return GdacsFloodExtraction
            case _:
                raise ValueError(f"No extractor for connector type: {self.get_type_display()}")

    
```